# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/stat-nsec.patch
# Copyright (C) 2007 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- ./libc/sysdeps/linux/common/xstatconv.c.orig	2006-08-30 09:10:25.000000000 +0200
+++ ./libc/sysdeps/linux/common/xstatconv.c	2006-12-01 16:04:01.000000000 +0100
@@ -39,16 +39,18 @@
 	buf->st_blocks = kbuf->st_blocks;
 	buf->st_atime = kbuf->st_atime;
 	buf->st_mtime = kbuf->st_mtime;
 	buf->st_ctime = kbuf->st_ctime;
 #ifdef STAT_HAVE_NSEC
 	buf->st_atimensec = kbuf->st_atime_nsec;
 	buf->st_mtimensec = kbuf->st_mtime_nsec;
 	buf->st_ctimensec = kbuf->st_ctime_nsec;
+#else
+	buf->st_atimensec = buf->st_mtimensec = buf->st_ctimensec = 0;
 #endif
 }
 
 #ifdef __UCLIBC_HAS_LFS__
 
 void attribute_hidden __xstat64_conv(struct kernel_stat64 *kbuf, struct stat64 *buf)
 {
 	/* Convert to current kernel version of `struct stat64'. */
@@ -67,12 +69,14 @@
 	buf->st_blocks = kbuf->st_blocks;
 	buf->st_atime = kbuf->st_atime;
 	buf->st_mtime = kbuf->st_mtime;
 	buf->st_ctime = kbuf->st_ctime;
 # ifdef STAT_HAVE_NSEC
 	buf->st_atimensec = kbuf->st_atime_nsec;
 	buf->st_mtimensec = kbuf->st_mtime_nsec;
 	buf->st_ctimensec = kbuf->st_ctime_nsec;
+#else
+	buf->st_atimensec = buf->st_mtimensec = buf->st_ctimensec = 0;
 # endif
 }
 
 #endif /* __UCLIBC_HAS_LFS__ */
